hw2_decor_iter/HW_2.py

```diff
@@ -0,0 +1,337 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# # Задание 1 (2 балла)
+
+# Напишите класс `MyDict`, который будет полностью повторять поведение обычного словаря, за исключением того, что при итерации мы должны получать и ключи, и значения.
+# 
+# **Модули использовать нельзя**
+
+# In[22]:
+
+
+class MyDict(dict):
+    def __iter__(self):
+        return zip(self.keys(), self.values())
+
+
+# In[23]:
+
+
+dct = MyDict({"a": 1, "b": 2, "c": 3, "d": 25})
+dct
+
+
+# In[33]:
+
+
+for item in iter(dct):
+    print(item)
+
+
+# In[29]:
+
+
+for key, value in dct:
+    print(key, value)   
+
+
+# In[30]:
+
+
+for key, value in dct.items():
+    print(key, value)
+
+
+# In[31]:
+
+
+for key in dct.keys():
+    print(key)
+
+
+# In[32]:
+
+
+dct["c"] + dct["d"]
+
+
+# # Задание 2 (2 балла)
+
+# Напишите функцию `iter_append`, которая "добавляет" новый элемент в конец итератора, возвращая итератор, который включает изначальные элементы и новый элемент. Итерироваться по итератору внутри функции нельзя, то есть вот такая штука не принимается
+# ```python
+# def iter_append(iterator, item):
+#     lst = list(iterator) + [item]
+#     return iter(lst)
+# ```
+# 
+# **Модули использовать нельзя**
+
+# In[58]:
+
+
+def iter_append(iterator, item):
+    new_iterator = []
+    while True:
+        try:
+            elem = next(iterator)
+            new_iterator.append(elem)
+        except StopIteration:
+            new_iterator.append(item)
+            break
+    return new_iterator
```

Comment:
![image](https://user-images.githubusercontent.com/48160953/228952111-0324cdbc-9340-4303-9f20-82506139f100.png)
```suggestion
def iter_append(iterator, item):
    yield from iterator
    yield item      
```

Reply:
Список это не итератор, а iterable